File: perfomance-json-parser/src/main/java/com/steelzack/testing/jsonparser/ParserJSON.java
package com.steelzack.testing.jsonparser;

import java.awt.Color;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import java.util.Map.Entry;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class ParserJSON {

	private Map<String, int[]> colorMap = null;

	public ParserJSON(String fileName) {
		JSONParser parser = new JSONParser();

		try {

			Object obj = parser.parse(new FileReader(fileName));

			JSONObject jsonObject = (JSONObject) obj;

			Iterator<?> i = jsonObject.entrySet().iterator();

			Map<String, int[]> colorMap = new HashMap<String, int[]>();
			while (i.hasNext()) {
				Entry<?, ?> colorObject = (Entry<?, ?>) i.next();
				Color color = Color.decode(colorObject.getValue().toString());
				colorMap.put(
						colorObject.getKey().toString(),
						new int[] { color.getRed(), color.getGreen(),
								color.getBlue() });
			}

			this.colorMap = colorMap;

		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		} catch (ParseException e) {
			e.printStackTrace();
		}

	}

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		ParserJSON obj = new ParserJSON("c:\\colours.json");
		System.out.println(obj.getColour("white", 0.75f));
		System.out.println(obj.toJSON(0.75f));
	}

	public String getColour(String name, float brightness) {
		int[] rgbValues = this.colorMap.get(name);
		float[] hsbVals = Color.RGBtoHSB(rgbValues[0], rgbValues[1],
				rgbValues[2], null);
		Color changedColor = Color.getHSBColor(hsbVals[0], hsbVals[1],
				brightness * hsbVals[2]);
		this.colorMap.put(name,
				new int[] { changedColor.getRed(), changedColor.getGreen(),
						changedColor.getBlue() });
		
		String hexString = Integer.toHexString(changedColor.getRGB() & 0x00ffffff);
		String zeroes = "000000";
		hexString = zeroes.substring(hexString.length()).concat(hexString);
		return "#"
				.concat(hexString);
	}

	@SuppressWarnings("unchecked")
	public String toJSON(float brightness) {
		Iterator<Entry<String, int[]>> i = this.colorMap.entrySet().iterator();
		JSONObject jsonObject = new JSONObject();
		HashMap<String, int[]> newColorMap = new HashMap<String, int[]>();
		
		do {
			Entry<String, int[]> object = (Entry<String, int[]>) i.next();
			newColorMap.put(object.getKey(), object.getValue());
		} while (i.hasNext());

		i = newColorMap.entrySet().iterator();

		do {
			Entry<String, int[]> object = (Entry<String, int[]>) i.next();
			String hexString = getColour(object.getKey(), brightness);
			jsonObject.put(object.getKey(), hexString);
		} while (i.hasNext());

		return jsonObject.toJSONString();
	}
}

File: json-performance/perfomance-json-parser/src/main/kotlin/org/jesperancinha/testing/jsonparser/ParserJSON.kt
package org.jesperancinha.testing.jsonparser

import org.json.simple.JSONObject
import org.json.simple.parser.JSONParser
import org.json.simple.parser.ParseException
import java.awt.Color
import java.io.FileReader
import java.io.IOException

class ParserJSON(fileName: String) {
    private val colorMap: MutableMap<String, IntArray> by lazy {
        val parser = JSONParser()
        val obj = parser.parse(FileReader(fileName))
        val jsonObject = obj as JSONObject
        val i: Iterator<*> = jsonObject.entries.iterator()
        val colorMap: MutableMap<String, IntArray> = HashMap()
        while (i.hasNext()) {
            val (key, value) = i.next() as Map.Entry<*, *>
            val color = Color.decode(value.toString())
            colorMap[key.toString()] = intArrayOf(
                color.red, color.green,
                color.blue
            )
        }
        colorMap
    }

    fun getColour(name: String, brightness: Float): String {
        val rgbValues = colorMap[name]
       return rgbValues?.let {
            val hsbVals = Color.RGBtoHSB(
                rgbValues[0], rgbValues[1],
                rgbValues[2], null
            )
            val changedColor = Color.getHSBColor(
                hsbVals[0], hsbVals[1],
                brightness * hsbVals[2]
            )
            colorMap[name] = intArrayOf(
                changedColor.red, changedColor.green,
                changedColor.blue
            )
            var hexString = Integer.toHexString(changedColor.rgb and 0x00ffffff)
            val zeroes = "000000"
            hexString = zeroes.substring(hexString.length) + hexString
            "#$hexString"
        } ?: "#000000"
    }

    fun toJSON(brightness: Float): String {
        var i: Iterator<Map.Entry<String, IntArray>> = colorMap.entries.iterator()
        val jsonObject = JSONObject()
        val newColorMap = HashMap<String, IntArray>()
        do {
            val (key, value) = i.next()
            newColorMap[key] = value
        } while (i.hasNext())
        i = newColorMap.entries.iterator()
        do {
            val (key) = i.next()
            val hexString = getColour(key, brightness)
            jsonObject[key] = hexString
        } while (i.hasNext())
        return jsonObject.toJSONString()
    }

    companion object {
        /**
         * @param args
         */
        @JvmStatic
        fun main(args: Array<String>) {
            val obj = ParserJSON("c:\\colours.json")
            println(obj.getColour("white", 0.75f))
            println(obj.toJSON(0.75f))
        }
    }
}